server_tests/test/clientTests.js

```diff
@@ -1201,13 +1242,13 @@ describe("RUST API TEST SUITE", function() {
             var deferred = Q.defer();
 
             return Q.all([testClient.takePicture(sessionId, function(res) {
-                return testClient.bublPoll(res.body.id, '', 'wrongtype')
+                testClient.bublPoll(res.body.id, '', 'wrongtype')
                 .then( expectError,
                     (err) => {Comparison.invalidParameterValueError(err);
                     deferred.resolve();
                 });
             }).then( function onSuccess (res) {
-                Comparison.oscTakePictureOutput(res);
+                (res) => Comparison.oscTakePictureOutput(res);
```

Comment:
.reject() "resolves" a rejected promise correct? If deferred.reject() at line 1248, the promise created would have a fulfilled state for the .then at line 1250?